docs/worldgen/biomemodifier.md

```diff
@@ -0,0 +1,741 @@
+
+<Tabs>
+  <TabItem value="json" label="JSON" default>
+
+```json5
+{
+    "type": "neoforge:remove_spawns",
+
+    // Can either be an id "minecraft:plains"
+    // List of ids ["minecraft:plains", "minecraft:badlands", ...]
+    // Or a tag "#c:is_overworld"
+    "biomes": "#namespace:biome_tag",
+
+    // Can either be an id "minecraft:ghast"
+    // List of ids ["minecraft:ghast", "minecraft:skeleton", ...]
+    // Or a tag "#minecraft:skeletons"
+    "entity_types": "#namespace:entitytype_tag"
+}
+```
+
+  </TabItem>
+  <TabItem value="datagen" label="Datagen">
+
+```java
+// Define keys for Datapack registry objects
+
+public static final ResourceKey<BiomeModifier> REMOVE_SPAWNS_EXAMPLE =
+    ResourceKey.create(
+        NeoForgeRegistries.Keys.BIOME_MODIFIERS, // The registry this key is for
+        ResourceLocation.fromNamespaceAndPath(MOD_ID, "remove_spawns_example") // The registry name
+    );
+
+// For some RegistrySetBuilder BUILDER
+// being passed to DatapackBuiltinEntriesProvider
+// in a listener for GatherDataEvent
+BUILDER.add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, bootstrap -> {
+    // Lookup any necessary registries
+    // Static registries only need to be looked up if you need to grab the tag data
+    HolderGetter<Biome> biomes = bootstrap.lookup(Registries.BIOME);
+    HolderGetter<EntityType<?>> entities = bootstrap.lookup(Registries.ENTITY_TYPE);
+
+    // Register the Biome Modifiers
+
+    bootstrap.register(REMOVE_SPAWNS_EXAMPLE,
+        new RemoveSpawnsBiomeModifier(
+            // The biome(s) to remove the spawns from
+            biomes.getOrThrow(Tags.Biomes.IS_OVERWORLD),
+            // The entities to remove spawns for
+            entities.getOrThrow(EntityTypeTags.SKELETONS)
+        )
+    );
+})
+```
+
+  </TabItem>
+</Tabs>
+
+
+### Add Spawn Costs
+
+Allows for adding new Spawn Costs to biomes. Spawn Costs are a newer way of making mobs spawn spread out in a biome to reduce clustering. It works by having the entities give off a `charge` that surrounds them and adds up with other entity's `charge`. Then when spawning, it looks for a spot where the total `charge` field at the location multiplied by the spawning entity's `charge` value is less than the spawning entity's `energy_budget`. This is an advanced way of spawning mobs so it is a good idea to reference the Soul Sand Valley Biome for existing values to borrow.
+
+The modifier takes in the `Biome` id or tag of the biomes the spawn costs are added to, the `EntityType` id or tag of the mobs to add spawn costs for, and the `MobSpawnSettings.MobSpawnCost` of the mob. The `MobSpawnCost` contains the energy budget, which indicates the maximum number of entities that can spawn in a location based upon the charge provided for each entity spawned.
+
+**NOTE:** If you are a modder adding a new mob, make sure the mob has a spawn restriction registered to RegisterSpawnPlacementsEvent event.
```

Comment:
Why? This does not affect anything as the modder can choose whether to have restrictions placed on spawning or not.

Reply:
If you’re adding a nea mob to the world, you need to register spawn restrictions or else the mob iirc, has no restriction and will spawn in midair, fall, and die and you end up confused as to why the biome doesn’t have your mob. In fact, neoforge will throw a dev log warning if it finds a mob in a biome without a registered spawn restriction. 

I can add one more sentence to state it is to make sure the mob spawns on ground and such

Reply:
Well yes, but a modder may want to create a mob that noclips through the ground. That's why I wasn't a big fan of the wording. But yeah, something to indicate for the common mob is fine.